src/module/fluid.js

```diff
@@ -38,9 +38,13 @@ var upPath = path.resolve(__dirname, "../../../../..");
 var upInfusionPath = resolveModuleSync("infusion", upPath);
 if (upInfusionPath) {
     upInfusion = require(upInfusionPath);
-    upInfusion.log("Resolved infusion from path " + __dirname + " to " + upInfusion.module.modules.infusion.baseDir);
-    module.exports = upInfusion;
-    return;
+    if (upInfusion && Object.keys(upInfusion).length !== 0) {
```

Comment:
i) This check should have a substantial comment including a) the JIRA number, b) a rough description of the situation the problem occurs in and c) its mechanism of action
ii) A check for whether upInfusion.module is defined or not is a more conservative check